src/common/utils/auth_tests.rs

```diff
@@ -0,0 +1,977 @@
+            AuthExtractor {
+                auth: format!("{AUTH_HEADER_VAL}"),
+                method: format!("{GET_METHOD}"),
+                o2_type: format!("_data_stream:{ORG_ID}"),
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn create_org_data_stream() {
+        test_auth(
+            Method::PUT,
+            format!("api/{ORG_ID}/_data_stream/NAME"),
+            AuthExtractor {
+                auth: format!("{AUTH_HEADER_VAL}"),
+                method: format!("{PUT_METHOD}"),
+                o2_type: format!("_data_stream:NAME"), // Is this correct?
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn get_org_pipeline() {
+        test_auth(
+            Method::GET,
+            format!("api/{ORG_ID}/_ingest/pipeline/NAME"),
+            AuthExtractor {
+                auth: format!("{AUTH_HEADER_VAL}"),
+                method: format!("{GET_METHOD}"),
+                o2_type: format!("_ingest:pipeline"), // Is this correct?
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn create_org_pipeline() {
+        test_auth(
+            Method::PUT,
+            format!("api/{ORG_ID}/_ingest/pipeline/NAME"),
+            AuthExtractor {
+                auth: format!("{AUTH_HEADER_VAL}"),
+                method: format!("{PUT_METHOD}"),
+                o2_type: format!("pipeline:NAME"), // Is this correct?
+                org_id: format!("{ORG_ID}"),
+                bypass_check: false,
+                parent_id: format!("default"),
+            },
+        )
+        .await
+    }
+
+    /// Tests that the correct authorization information is extracted from a
+    /// request to the given path.
+    async fn test_auth(method: Method, path: String, expected: AuthExtractor) {
+        use std::sync::Arc;
+
+        let mut config = mock_config();
+        config.common.web_url = API_ROOT_URI.to_string();
+        config.common.base_uri = "".to_string(); // API_BASE_COMP.to_string();
+        config::config::CONFIG.store(Arc::new(config));
+
+        let req = test::TestRequest::default()
+            .method(method)
+            .uri(&format!("{API_ROOT_URI}/{path}"))
+            .append_header(AUTH_HEADER)
+            .to_http_request();
+        let actual = AuthExtractor::extract(&req).await.unwrap();
+        assert_eq!(actual, expected);
+    }
+
+    /// Returns a mock configuration with fields set to default values.
+    #[allow(deprecated)]
+    fn mock_config() -> config::Config {
```

Comment:
Since the `AuthExtractor` accesses a global `config::Config` state in its logic, we need to mock up the entire structure for the tests.

If we could refactor `AuthExtractor` so that the logic we want to test accepts function parameters for each of the config fields/values that it needs, then to test that logic we could simply mock and pass in the config fields the logic needs to execute rather then the entire `config::Config` structure.